metricbeat/module/kubernetes/util/mapstr.go

```diff
@@ -0,0 +1,98 @@
+package util
+
+import (
+	"github.com/elastic/beats/libbeat/common"
+)
+
+// GetString from the given event, "" if it doesn't exist
+func GetString(m common.MapStr, field string) string {
+	val, err := m.GetValue(field)
+	if err != nil {
+		return ""
+	}
+
+	valStr, ok := val.(string)
+	if !ok {
+		return ""
+	}
+	return valStr
+}
+
+// GetFloat64 from the given event, 0 if it doesn't exist
+func GetFloat64(m common.MapStr, field string) float64 {
+	val, err := m.GetValue(field)
+	if err != nil {
+		return 0
+	}
+
+	valStr, ok := val.(float64)
+	if !ok {
+		return 0
+	}
+	return valStr
+}
+
+// GetInt64 from the given event, 0 if it doesn't exist
+func GetInt64(m common.MapStr, field string) int64 {
+	val, err := m.GetValue(field)
+	if err != nil {
+		return 0
+	}
+
+	valStr, ok := val.(int64)
+	if !ok {
+		return 0
+	}
+	return valStr
+}
+
+// MergeEvents from b events into a. The process will ensure that:
+//   - only events matching the given filter are processed.
+//   - fields in the delete list will be removed from the event
+//   - match fields will be used to match events from a & b, if all fields are equal, they will be merged
```

Comment:
Could we separate this into three functions? one for merging, another one for filtering and another one to delete fields?